### PR TITLE
[Bug] Fix themes module import causing crashes in va-mobile-app

### LINKED ISSUE #377: DS - Export useTheme hook
## Description
A `useTheme()` hook was created as part of #263 which returns the appropriate light or dark theme based on the device settings. We should make this hook available externally by our consumers so they can have the option to use the hook and more easily select the themes we've made available to them.

## Acceptance Criteria
- [ ] Export the `useTheme()` hook in `index.tsx`
- [ ] Test that the hook is functional in the app

### PR DESCRIPTION
**Pull request by narin**
## Description of Change

- Fixes the `TypeError: Cannot read property 'themes' of undefined` error we are seeing in the app by using a namespace import instead of destructured import in our hooks file.
- Set tokens and assets packages as peerDependencies and devDependencies in the components package.json as the app is now adding tokens and assets directly in their dependencies
- Exports the `useTheme()` hook which I've tested in the app and closes #377. 

I could not find a way to keep the destructured import after several attempts of refactoring our tokens exports. I think the namespace import is an acceptable workaround for now.

Things I've tried without success:
- Combining colors and themes exports into `index.js` instead of having them as separate `colors.js` and `themes.js`
- Combining `types/colors.ts` and `types/themes.ts` into `index.d.ts`
- Declaring the `@department-of-veterans-affairs/mobile-tokens` module in `index.d.ts` and wrapping it around the color/themes types
- Moving `index.d.ts` to same folder as `index.js`
- Using commonJS exports instead
- Setting `"type": "module"` in `package.json` and converting style dictionary config files to `.cjs`
- Setting `exports` in `package.json`
- Adjusting several settings in va-mobile-app's `tsconfig.json`

I've created a follow-up ticket #390 to determine root cause when we have additional bandwidth.

#### Testing Packages

- components@0.19.2-alpha.0
- components@0.19.2-alpha.1
- components@0.19.2-alpha.2
- components@0.19.2-alpha.3
- components@0.19.2-alpha.4
- components@0.19.2-alpha.5

## Testing

- [x] Tested on iOS 
- [x] Tested on Android 
- [x] Tested on Web

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish

If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy) and the PR is approved and ready to merge:
- [x] Merge `main` into branch
- [ ] Merge branch to `main`
- [ ] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [ ] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

### CODE REVIEW

**Comment by TimRoe** [comment]
> Screenshots/Video

Just as a sanity check: your going into the app did go deeper than the login screen, right? It shouldn't matter since `main` and correspondingly your alpha test branches should have theme used by the Button component, but given how it slipped by the first time, would be good to have confirmation that screens containing Links/Segmented controls were also accessed, didn't crash, and toggled light/dark mode as expected.

**Comment by narin** [comment]
> > Screenshots/Video
> 
> Just as a sanity check: your going into the app did go deeper than the login screen, right? It shouldn't matter since `main` and correspondingly your alpha test branches should have theme used by the Button component, but given how it slipped by the first time, would be good to have confirmation that screens containing Links/Segmented controls were also accessed, didn't crash, and toggled light/dark mode as expected.

Yup, went through the sections of the app with SegmentedControl and Link as well. Toggling light/dark mode works as expected. Our components do seem to toggle between light/dark mode faster than the rest of the app but I think that has always been the case.

File: packages/components/src/index.tsx [hunk 1]
```diff
@@ -14,3 +14,4 @@ export { Link } from './components/Link/Link'
 export { LoadingIndicator } from './components/LoadingIndicator/LoadingIndicator'
 export { SegmentedControl } from './components/SegmentedControl/SegmentedControl'
 export { Spacer } from './components/Spacer/Spacer'
+export { useTheme } from './utils'
```
Comment (TimRoe) [inline packages/components/src/index.tsx:17]: Nitpicky, but should we split out a separate section (e.g. `// Export consumer available utilities here so they are exported through npm`) for non-component exports? 

Seems like a good idea to keep some order moving forward as the components list grows and potentially other utilities join the exports.

**Review by TimRoe — APPROVED** [review]
Approved with one minor suggestion (that could always be done later).

**Comment by TimRoe** [comment]
> Our components do seem to toggle between light/dark mode faster than the rest of the app but I think that has always been the case.

It has, the useColorScheme hook works faster than the app theme wrapper.

**Pull request merged**